src/controller/contracts.ts

```diff
@@ -0,0 +1,55 @@
+import { Request, Response, NextFunction } from 'express';
+import { Contract } from '../classes/contracts';
+
+// get all contracts
+const getContracts = async (req: Request, res: Response, next: NextFunction) => {
+    let contract = new Contract();
+    return res.status(200).json(await contract.readAll())
+};
+
+// get one contract
+const getContract = async (req: Request, res: Response, next: NextFunction) => {
+    let contract = new Contract();
+    return res.status(200).json(await contract.readContract(Number(req.params.id)));
+};
+
+// update contract
+const updateContract = async (req: Request, res: Response, next: NextFunction) => {
+    let contract = new Contract(req.body.StartDate, req.body.EndDate, Number(req.body.ClientID), req.body.ClientType, Number(req.body.AdvancedPayement), Number(req.body.Price), Number(req.body.ContractID));
```

Comment:
Casting your body properties like `Price` to a number will not throw an error if it is not a valid number, instead it will return `NaN` (not a number). Your validators later on do not check if the value is NaN and thus will accept any value given in the body